apps/src/music/views/PatternAiPanel.tsx

```diff
@@ -112,6 +227,22 @@ const PatternAiPanel: React.FunctionComponent<PatternAiPanelProps> = ({
     onChange(currentValue);
   };
 
+  const getOuterCellClasses = (tick: number) => {
+    return classNames(
+      styles.outerCell,
+      tick === currentPreviewTick &&
+        generateState === 'none' &&
+        styles.outerCellPlaying,
+      generateState === 'generating' &&
+        tick === generatingScanStep &&
+        styles.outerCellScanning,
+      generateState === 'generating' &&
+        tick !== generatingScanStep &&
+        styles.outerCellSlowFade,
+      tick % 4 === 0 && styles.outerCellFourth
```

Comment:
minor: can use `:nth-child` for this?

Reply:
I do like CSS when possible, but given the amount of styling already done by code here, I think it's best to do it all in one place, rather than having two different systems that might collide with each other.